marklogic-client-api-functionaltests/src/test/java/com/marklogic/client/fastfunctest/TestSearchPageLength.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © 2024 MarkLogic Corporation. All Rights Reserved.
+ */
+
+package com.marklogic.client.fastfunctest;
+
+import com.marklogic.client.impl.OkHttpServices;
+import com.marklogic.client.impl.RESTServices;
+import com.marklogic.client.io.*;
+import com.marklogic.client.query.*;
+import org.custommonkey.xmlunit.exceptions.XpathException;
+import org.junit.jupiter.api.*;
+import org.w3c.dom.Document;
+
+import static org.custommonkey.xmlunit.XMLAssert.*;
+
+
+class TestSearchPageLength extends AbstractFunctionalTest {
+
+	static final QueryManager queryMgr = client.newQueryManager();
+	StringQueryDefinition qd = queryMgr.newStringDefinition();
+
+	@Test
+	void testSearchPageLengthZero() throws XpathException {
+		queryMgr.setPageLength(0);
+		qd.setCriteria("Foo");
+		DOMHandle resultsHandle = new DOMHandle();
+		queryMgr.search(qd, resultsHandle);
+
+		Document resultDoc = resultsHandle.get();
+		assertXpathEvaluatesTo("0", "string(//*[local-name()='response']//@*[local-name()='page-length'])", resultDoc);
+	}
+
+
+	@Test
+	void testSearchPageLengthNegative() throws XpathException {
+		queryMgr.setPageLength(-1);
+		qd.setCriteria("Foo");
+		DOMHandle resultsHandle = new DOMHandle();
+		queryMgr.search(qd, resultsHandle);
+
+		Document resultDoc = resultsHandle.get();
+		assertXpathEvaluatesTo("10", "string(//*[local-name()='response']//@*[local-name()='page-length'])", resultDoc);
+	}
+
+	@Test
+	void testOkHttpPageLengthParamaterZero() throws XpathException {
```

Comment:
We don't need this test or the one below it - this isn't a public API for our users. Only need to test `queryManager.search`.